Fable.Import.NodeLibzfs/NodeLibzfs.fs

```diff
@@ -1,70 +1,88 @@
-// ts2fable 0.1.9
+// Copyright (c) 2018 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
 module rec libzfs
 open Fable.Core
 
 let [<Import("default","@iml/node-libzfs")>] libzfs: Libzfs.IExports = jsNative
 
 module Libzfs =
-
     type [<AllowNullLiteral>] IExports =
         [<Emit("$0()")>] abstract Invoke: unit -> NodeLibzfs
 
-    type [<AllowNullLiteral>] Dataset =
-        abstract name: string with get, set
-        abstract kind: string with get, set
+    type [<AllowNullLiteral>] RootNode =
+        abstract Root: Root with get, set
+
+    type [<AllowNullLiteral>] Root =
+        abstract children: VDev list with get, set
+
+    type [<AllowNullLiteral>] FileNode =
+        abstract File: File with get, set
 
     type [<AllowNullLiteral>] File =
+        abstract guid: string option with get, set
+        abstract state: string with get, set
         abstract path: string with get, set
+        abstract is_log: bool option with get, set
+
+    type [<AllowNullLiteral>] DiskNode =
+        abstract Disk: Disk with get, set
 
     type [<AllowNullLiteral>] Disk =
+        abstract guid: string option with get, set
```

Comment:
should this be an option?

Reply:
Right now it's modeled as an option in: https://github.com/intel-hpdd/rust-libzfs/blob/3210dbf130596089fe0d04c9b83cf0c6d9ef6a1d/libzfs/src/lib.rs#L349-L353

as it's looking up a prop in an nvlist. I imagine that guid will always be there, but let's get a little more usage to prove it out and then we can unwrap it.